lib/community/solicitor.js

```diff
@@ -46,13 +57,17 @@ var queueEmail = function (emailData) {
 }
 
 Solicitor.sendToUser = function (user, community) {
-  return queueEmail(emailData(user, community))
+  if (user.get('settings').email_solicitation_preference) {
+    return queueEmail(emailData(user, community))
+  } else {
+    return Promise.resolve()
+  }
```

Comment:
Another way to do this is to wrap the entire method in `Promise.method` and then just return nothing if the user doesn't have the setting turned on